Python/Product/Cookiecutter/Model/CookiecutterClient.cs

```diff
@@ -88,24 +88,34 @@ class CookiecutterClient : ICookiecutterClient {
             var contextJson = string.Join(Environment.NewLine, result.StandardOutputLines);
             var context = (JToken)JObject.Parse(contextJson).SelectToken("cookiecutter");
             if (context != null) {
+                JProperty vsExtrasProp = null;
+
                 foreach (JProperty prop in context) {
                     // Properties that start with underscore are for internal use,
                     // and cookiecutter doesn't prompt for them.
                     if (!prop.Name.StartsWith("_")) {
-                        if (prop.Value.Type == JTokenType.String || prop.Value.Type == JTokenType.Integer || prop.Value.Type == JTokenType.Float) {
-                            items.Add(new ContextItem(prop.Name, prop.Value.ToString()));
+                        if (prop.Value.Type == JTokenType.String) {
+                            items.Add(new ContextItem(prop.Name, ContextItemValueType.String, prop.Value.ToString()));
+                        } else if (prop.Value.Type == JTokenType.Integer || prop.Value.Type == JTokenType.Float) {
+                            items.Add(new ContextItem(prop.Name, ContextItemValueType.Numeric, prop.Value.ToString()));
                         } else if (prop.Value.Type == JTokenType.Array) {
                             var elements = new List<string>();
                             JArray ar = prop.Value as JArray;
                             foreach (JToken element in ar) {
                                 elements.Add(element.ToString());
                             }
-                            items.Add(new ContextItem(prop.Name, elements[0], elements.ToArray()));
+                            items.Add(new ContextItem(prop.Name, ContextItemValueType.List, elements[0], elements.ToArray()));
                         } else {
                             throw new InvalidOperationException(string.Format("Unsupported json element type in context file for property '{0}'.", prop.Name));
                         }
+                    } else if (prop.Name == "_visual_studio") {
+                        vsExtrasProp = prop;
                     }
                 }
+
+                if (vsExtrasProp != null) {
+                    LoadVisualStudioSpecificContext(items, vsExtrasProp);
```

Comment:
Is there some way we can report errors in this section without aborting the entire template? Just fall back to regular behaviour